### DISCUSSION TITLE
Architecture dependent requirements

### DISCUSSION THREAD

**Discussion by gt6989b** [python-poetry/poetry#8448] [Q&A] [answered]
I am supporting a system which is main running Python 3.10 on Windows but has limited delivery on Linux. We use `poetry` to manage the package environment.

For NTLM authentication with requests, I introduced a dependency on the `requests_negotiate_sspi` package, which brings in `pywin32`. When I run this on Windows, everything works, but when I try `poetry install` on Linux, I get a
```
RuntimeError: Unable to find installation candidates for pywin32 (306)
```
If I understand correctly, this is an issue since no such package exists for Linux, which makes sense. I don't execute this dependency on Linux.

Could I make a conditional dependency in `poetry`, where it would only depend on this package when running under Windows, but not on Linux? If not, what other options do I have to deal with this issue?

Thank you very much.

**Comment by clintonroy** [comment, marked as answer]
https://python-poetry.org/docs/dependency-specification#using-environment-markers
